src/scandeval/text_to_text.py

```diff
@@ -0,0 +1,236 @@
+            id2label (list of str):
+                Conversion of indices to labels.
+
+        Returns:
+            A dictionary with the names of the metrics as keys and the metric values as
+            values.
+        """
+        model_outputs, labels = model_outputs_and_labels
+
+        model_output_dtype = np.asarray(model_outputs).dtype
+        if model_output_dtype in [np.float16, np.float32, np.float64]:
+            predictions = np.asarray(model_outputs).argmax(axis=-1)
+        else:
+            predictions = model_outputs
+
+        results: dict[str, float] = dict()
+        for cfg in self.dataset_config.task.metrics:
+            metric = self._metrics[cfg.name]
+            score_dict: dict[str, float] | None = metric.compute(
+                predictions=predictions,
+                references=labels,
+                **cfg.compute_kwargs,
+            )
+            if score_dict is not None:
+                scores = score_dict[cfg.results_key]
+                if isinstance(scores, list):
+                    scores = sum(scores) / len(scores)
+                results[cfg.name] = scores
+        return results
+
+    def _extract_few_shot_examples(
+        self, train_dataset: Dataset, random_seed: int
+    ) -> list[dict[str, Any]]:
+        """Extract few-shot examples from the training dataset.
+
+        Args:
+            train_dataset:
+                The training dataset.
+            random_seed:
+                The random seed to use when extracting the few-shot examples.
+
+        Returns:
+            The few-shot examples.
+        """
+        shuffled_train = train_dataset.shuffle(seed=random_seed)
+        num_few_shots = self.dataset_config.num_few_shot_examples
+        few_shot_examples: list[dict[str, Any]] = list()
+        while len(few_shot_examples) < num_few_shots:
+            example = shuffled_train.select(range(1))[0]
+            few_shot_examples.append(example)
+            shuffled_train = shuffled_train.filter(
+                lambda x: x["text"] != example["text"]
+            )
+        random.seed(random_seed)
+        random.shuffle(few_shot_examples)
+        return few_shot_examples
+
+    def _apply_few_shot_prompt(
+        self, examples: dict, few_shot_examples: list[dict]
+    ) -> dict:
+        """Apply a few-shot prompt to the examples.
+
+        Args:
+            examples:
+                The examples to apply the prompt to.
+            few_shot_examples:
+                The examples to be included in the few-shot prompt.
+
+        Returns:
+            The examples with the few-shot prompt applied.
+        """
+        # Build the few-shot part of the prompt
+        few_shot_prompts = [
+            self.dataset_config.prompt_template.format(
+                text=example["text"].replace("\n", " ").strip(),
+                target_text=example["target_text"].replace("\n", " ").strip(),
+            )
+            for example in few_shot_examples
+        ]
```

Comment:
Would probably make it a function:
```suggestion
        few_shot_prompts = [self.format_prompt(example["text"], example["target_text"])
        for example in few_shot_examples]
```

Reply:
But it's already a function, no? More specifically, the `self.dataset_config.prompt_template.format` function? It seems like all this does is renaming that function to `format_prompt`?